src/java.base/macosx/native/libjava/java_props_macosx.c

```diff
@@ -247,15 +247,34 @@ void setOSNameAndVersion(java_props_t *sprops) {
         [invoke getReturnValue:&osVer];
 
         NSString *nsVerStr;
-        if (osVer.patchVersion == 0) { // Omit trailing ".0"
-            nsVerStr = [NSString stringWithFormat:@"%ld.%ld",
-                    (long)osVer.majorVersion, (long)osVer.minorVersion];
-        } else {
-            nsVerStr = [NSString stringWithFormat:@"%ld.%ld.%ld",
-                    (long)osVer.majorVersion, (long)osVer.minorVersion, (long)osVer.patchVersion];
+        // Copy out the char* if running on version other than pre-10.16 Mac OS (10.16 == 11.x)
```

Comment:
Isn't the comment backwards from what the test does? I would think it should be "if running on version other than 10.16 Mac OS ...". Or am I missing something?

Reply:
@kevinrushforth I think you are correct. This is actually mea culpa I think as I had provided in a Slack thread a failed attempt at a patch for this which contained this comment.

Reply:
So the words "other than" are too subtle?

Reply:
It's a double negative, unless I'm reading it incorrectly: "other than pre-10.16" I interpret as "not pre-10.16" or "10.16".

Reply:
`// Copy out the char* if running on version 10.x, where x < 16` ?

Reply:
It will also do it if running on `11.x[.y]` (once Xcode is upgraded), right?